src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -5017,10 +5022,25 @@ namespace ZoneTempPredictorCorrector {
 				ZoneMassFlowRate += Node( ZoneEquipConfig( ZoneEquipConfigNum ).InletNode( NodeNum ) ).MassFlowRate / ZoneMult;
 			} // NodeNum
 
+			ExhaustNodeFlag = false;
 			for ( NodeNum = 1; NodeNum <= ZoneEquipConfig( ZoneEquipConfigNum ).NumExhaustNodes; ++NodeNum ) {
-				ExhMassFlowRate += Node( ZoneEquipConfig( ZoneEquipConfigNum ).ExhaustNode( NodeNum ) ).MassFlowRate / ZoneMult;
+				ExhaustListFlag = false;
+				for ( iList = 1; iList <= ZoneEquipList( ZoneNum ).NumOfEquipTypes; ++iList ) {
+					if ( ( ZoneEquipList( ZoneNum ).EquipType_Num( iList ) == ZoneExhaustFan_Num ) && 
+						( ZoneEquipList( ZoneNum ).EquipData( iList ).NumInlets > 0 && ZoneEquipList( ZoneNum ).EquipData( iList ).InletNodeNums( 1 ) == ZoneEquipConfig( ZoneEquipConfigNum ).ExhaustNode( NodeNum ) ) ) {
+						if ( !( ZoneEquipConfig( ZoneEquipConfigNum ).AirLoopNum > 0 || ZoneAirMassFlow.EnforceZoneMassBalance || AirflowNetworkNumOfExhFan > 0 ) ) {
+							ExhaustNodeFlag = true;
+							ExhaustListFlag = true;
+						}
+					}
+				}
+				if ( !ExhaustListFlag ) {
+					ExhMassFlowRate += Node( ZoneEquipConfig( ZoneEquipConfigNum ).ExhaustNode( NodeNum ) ).MassFlowRate / ZoneMult;
+				}
 			} // NodeNum
-			ExhMassFlowRate -= ZoneEquipConfig( ZoneEquipConfigNum ).ZoneExhBalanced; // Balanced exhaust flow assumes there are other flows providing makeup air such as mixing or infiltration, so subtract it here
+			if ( !ExhaustNodeFlag ) {
+				ExhMassFlowRate -= ZoneEquipConfig( ZoneEquipConfigNum ).ZoneExhBalanced; // Balanced exhaust flow assumes there are other flows providing makeup air such as mixing or infiltration, so subtract it here
+			}
 
 			if ( ZoneEquipConfig( ZoneEquipConfigNum ).ReturnAirNode > 0 ) {
 				TotExitMassFlowRate = ExhMassFlowRate + Node( ZoneEquipConfig( ZoneEquipConfigNum ).ReturnAirNode ).MassFlowRate / ZoneMult;
```

Comment:
@lgu1234 I ran the three defect files. The results are different from plain v8.7, but the three files still produce results that are different from each other.

If this fix proves to be correct, you don't need to add the extra loop here.  We already have a variable for ZoneEquipConfig( ZoneEquipConfigNum ).ZoneExh which is the sum of the flows into all exhaust fans in the zone.  So, I think the equivalent to your fix would be to revert to original code and simply replace 
```
ExhMassFlowRate -= ZoneEquipConfig( ZoneEquipConfigNum ).ZoneExhBalanced;
```
with
```
ExhMassFlowRate -= ZoneEquipConfig( ZoneEquipConfigNum ).ZoneExh;
```
This would exclude all exhaust fan mass flows from ExhMassFlowRate regardless of balanced or unbalanced.